packages/core-orders/src/module/configureOrdersModule-queries.ts

```diff
@@ -202,6 +209,51 @@ export const configureOrdersModuleQueries = ({ Orders }: { Orders: mongodb.Colle
       });
       return !!orderCount;
     },
+
+    getOrderStatisticsRecordsByDate: async ({
+      field,
+      dateRange,
+    }: {
+      field: 'created' | 'confirmed' | 'ordered' | 'rejected';
+      dateRange?: DateFilterInput;
+    }): Promise<OrderStatisticsRecord[]> => {
+      const match: Record<string, any> = {};
+      if (dateRange?.start || dateRange?.end) {
+        match[field] = {};
+        if (dateRange?.start) match[field].$gte = new Date(dateRange.start);
+        if (dateRange?.end) match[field].$lte = new Date(dateRange.end);
+      } else {
+        match[field] = { $exists: true };
+      }
+
+      return (await Orders.aggregate([
+        { $match: match },
+        { $unwind: '$calculation' },
+        {
+          $group: {
+            _id: {
+              date: {
+                $dateToString: { format: '%Y-%m-%d', date: `$${field}` },
+              },
+              currency: '$currencyCode',
+            },
+            totalAmount: { $sum: '$calculation.amount' },
```

Comment:
That won't work, we can't use aggregate here, we should return Orders with the date field, calculation, currencyCode. That way you can accept further input to the total field and resolve accordingly:

```
total(order: OrderType, params: { category: string; useNetPrice: boolean }): Price {
    const pricing = OrderPricingSheet({
      calculation: order.calculation,
      currencyCode: order.currencyCode,
    });

    if (pricing.isValid()) {
      return pricing.total(params);
    }
    return null;
  },
```